runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollEmoteOverlay.java

```diff
@@ -0,0 +1,156 @@
+import java.awt.FontMetrics;
+import java.awt.Graphics2D;
+import java.awt.Point;
+import javax.inject.Inject;
+import net.runelite.api.Client;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.plugins.cluescrolls.clues.EmoteClue;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.TextComponent;
+
+public class ClueScrollEmoteOverlay extends Overlay
+{
+	private static final Color HIGHLIGHT_BACKGROUND = new Color(0, 255, 0, 100);
+
+	private final ClueScrollPlugin plugin;
+	private final Client client;
+	private final TextComponent textComponent = new TextComponent();
+
+	@Inject
+	public ClueScrollEmoteOverlay(ClueScrollPlugin plugin, Client client)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.plugin = plugin;
+		this.client = client;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics, Point parent)
+	{
+		Object clue = plugin.getClue();
+
+		if (clue == null)
+		{
+			return null;
+		}
+
+		if (clue instanceof EmoteClue)
+		{
+			EmoteClue emoteClue = (EmoteClue) clue;
+
+			if (!emoteClue.getFirstEmote().hasSprite())
+			{
+				return null;
+			}
+
+			Widget emoteContainer = client.getWidget(WidgetInfo.EMOTE_CONTAINER);
+
+			if (emoteContainer == null)
+			{
+				return null;
+			}
+
+			if (emoteContainer.isHidden())
+			{
+				return null;
+			}
+
+			Widget emoteWindow = client.getWidget(WidgetInfo.EMOTE_WINDOW);
+
+			if (emoteWindow == null)
+			{
+				return null;
+			}
+
+			for (Widget emoteWidget : emoteContainer.getDynamicChildren()) 
+			{
+				if (emoteWidget.getSpriteId() == emoteClue.getFirstEmote().getSpriteId())
+				{
+					highlightWidget(graphics, parent, emoteWindow, emoteWidget,
+							emoteClue.getSecondEmote() != null ? "1st" : "");
+				}
+				else if (emoteClue.getSecondEmote() != null
+					&& emoteWidget.getSpriteId() == emoteClue.getSecondEmote().getSpriteId())
+				{
+					highlightWidget(graphics, parent, emoteWindow, emoteWidget, "2nd");
+				}
+			}
+		}
+
+		return null;
+	}
+
+	private void highlightWidget(Graphics2D graphics, Point parent, Widget window, Widget widget, String text)
+	{
+		net.runelite.api.Point canvasLocation = widget.getCanvasLocation();
+
+		if (canvasLocation == null)
+		{
+			return;
+		}
+
+		// Don't draw outside the emotes window
+		net.runelite.api.Point windowLocation = window.getCanvasLocation();
+
+		if (windowLocation.getY() > canvasLocation.getY())
+		{
+			return;
+		}
+
+		if (windowLocation.getY() + window.getHeight() < canvasLocation.getY() + widget.getHeight())
+		{
+			return;
+		}
+
+		// Make the overlay background darker if user is hovering over it
+		net.runelite.api.Point mousePosition = client.getMouseCanvasPosition();
+
+		boolean hovering = mousePosition.getX() >= canvasLocation.getX()
```

Comment:
Constructing rectangle and using Rectangle#contains would probably reduce the logic required here slightly